### ISSUE TITLE
Delete category translation removes category

### ISSUE THREAD

**Issue by simsationX** [sulu/sulu#6300]
| Q | A
| --- | ---
| Sulu Version | 2.3.4
| PHP Version | PHP 8

# Actual Behavior

If you delete a category translation in a different locale, the category is deleted completely.

# Expected Behavior

If you delete a category translation in a different locale, the category with the default locale and its original category translation will stay.

# Steps to Reproduce

1) In https://sulu.rocks/admin/#/categories/ add a new category "football" in locale `en` 
2) Switch to locale `de` you will see category  "[en] football"
3) Add a category translation "fussball" for this category
4) Delete the german category translation "fussball"
5) Now category is deleted in locale `en` as well

Expected behaviour would be to see the "[en] football" again

# Possible Solutions

Remove `ON DELETE CASCADE` on the foreign key in category_translation

```
  CONSTRAINT `FK_5DCED5E3B8075882` FOREIGN KEY (`idCategories`) REFERENCES `ca_categories` (`id`) ON DELETE CASCADE,
  ```
  
 and use orphanRemoval?

**Comment by alexander-schranz** [comment]
There is currently no deletion implement for entities to remove one locale. If you remove an entity it will remove its [main entity](https://github.com/sulu/sulu/blob/61506048a84ae426ff99e2dc2d120c5285407854/src/Sulu/Bundle/CategoryBundle/Category/CategoryManager.php#L343) with all localizations. Only `pages` at current state have a toolbar action which is called `Delete current localization`. So if you want that behaviour that toolbar action should be added to the category to remove only the CategoryTranslation Entity and not the Category Entity.

The `ON DELETE CASCADE` is correctly: https://github.com/sulu/sulu/blob/61506048a84ae426ff99e2dc2d120c5285407854/src/Sulu/Bundle/CategoryBundle/Resources/config/doctrine/CategoryTranslation.orm.xml#L29
This says if the `Category` is removed remove all `CategoryTranslation`. This is different what cascade-remove on doctrine level means.

**Comment by simsationX** [comment]
Thank you for your reply.

The problem is even if I try to remove the CategoryTranslation entity programmatically:

```
        $translationEntity = $category->findTranslationByLocale($locale);
        $this->em->remove($translationEntity);
        $this->em->flush();
```

the main Category entity is deleted.

And if you try to remove the translation from the translations collection in the Category entity, it won't be persisted because the owning side is the CategoryTranslation.

In my opinion removing a translation in a different locale should be possible. The main entity should be removed when we don't have a category translation any more.

**Comment by alexander-schranz** [comment]
@ymc-sise That should definitly not happen. Can you test if your remove the following lines if it work like expected:

https://github.com/sulu/sulu/blob/61506048a84ae426ff99e2dc2d120c5285407854/src/Sulu/Bundle/CategoryBundle/Resources/config/doctrine/Category.orm.xml#L37-L39

As I think this should be replaced the following for meta and translations:

```xml
 <cascade> 
     <cascade-persist/> 
 </cascade> 
 ```

**Comment by alexander-schranz** [comment]
Sorry I think it is this lines which should be removed:

https://github.com/sulu/sulu/blob/61506048a84ae426ff99e2dc2d120c5285407854/src/Sulu/Bundle/CategoryBundle/Resources/config/doctrine/CategoryTranslation.orm.xml#L25-L27

https://github.com/sulu/sulu/blob/61506048a84ae426ff99e2dc2d120c5285407854/src/Sulu/Bundle/CategoryBundle/Resources/config/doctrine/CategoryMeta.orm.xml#L16-L18

And the category.orm.xml the cascade-all should be replaced with cascade-persist.

**Comment by alexander-schranz** [comment]
@ymc-sise do you want to create a PR for this if it works?

**Comment by niklasnatter** [comment]
Fixed in #6304
